src/contextprotector/mcp_json_cli.py

```diff
@@ -0,0 +1,977 @@
+            servers = self.config.get_servers()
+            updated_servers = servers.copy()
+        elif hasattr(self.config, "mcp_servers"):
+            # Legacy MCPJsonConfig
+            servers = self.config.mcp_servers
+            updated_servers = None
+        else:
+            servers = {}
+            updated_servers = None
+
+        for server_name in self.servers_to_wrap:
+            print(f"  🛡️  Wrapping {server_name}...")
+
+            try:
+                current_spec = servers[server_name]
+                wrapped_spec = current_spec.with_context_protector()
+
+                if updated_servers is not None:
+                    # New system
+                    updated_servers[server_name] = wrapped_spec
+                elif self.config and hasattr(self.config, "mcp_servers"):
+                    # Legacy system
+                    self.config.mcp_servers[server_name] = wrapped_spec
+
+                print(f"    ✓ Successfully wrapped {server_name}")
+                print(f"    New command: {wrapped_spec.command}")
+                if wrapped_spec.args:
+                    args_preview = " ".join(wrapped_spec.args)
+                    if len(args_preview) > MAX_ARGS_PREVIEW_LENGTH:
+                        args_preview = args_preview[:57] + "..."
+                    print(f"    New args: {args_preview}")
+
+            except ValueError as e:
+                print(f"    ❌ Failed to wrap {server_name}: {e}")
+
+        # Update the configuration with wrapped servers (new system only)
+        if updated_servers is not None:
+            self.config.set_servers(updated_servers)
+
+    def _save_with_confirmation(self) -> bool:  # noqa: PLR0911
+        """Show diff and ask for confirmation before saving."""
+        if not self.config:
+            return False
+
+        # Generate new JSON
+        if hasattr(self.config, "raw_data"):
+            # New unified config system
+            new_json = json.dumps(self.config.raw_data, indent=2)
+        elif hasattr(self.config, "to_dict"):
+            # Legacy MCPJsonConfig for backward compatibility
+            new_json = json.dumps(self.config.to_dict(), indent=2)
+        else:
+            return False
+
+        # Check if there are any changes (there should be since we wrapped servers)
+        if new_json.strip() == self.original_json.strip():
+            print("No changes detected. This shouldn't happen after wrapping.")
+            return False
+
+        # Show the diff
+        print("\n" + "=" * 70)
+        print("PROPOSED CHANGES")
+        print("=" * 70)
+
+        original_lines = self.original_json.splitlines(keepends=True)
+        new_lines = new_json.splitlines(keepends=True)
+
+        diff = difflib.unified_diff(
+            original_lines,
+            new_lines,
+            fromfile=f"{self.file_path} (original)",
+            tofile=f"{self.file_path} (wrapped)",
+            lineterm="",
+        )
+
+        for line in diff:
```

Comment:
I might be pendantic with this, but this looks identical to https://github.com/trailofbits/mcp-context-protector/pull/32/files#diff-59b4abaac37e87d1da83c6334b583a1a21715be6578e978b8fabae5b7d09f065R381